### PR TITLE
Perception: Fix cipv index; better visualization of 3D object position

### PR DESCRIPTION
**Pull request by techoe**

### CODE REVIEW

File: modules/perception/camera/app/cipv_camera.cc [hunk 1]
```diff
@@ -764,7 +773,7 @@ bool Cipv::DetermineCipv(const std::vector<base::LaneLine> &lane_objects,
       AINFO << "objects[" << i << "]->track_id: " << (*objects)[i]->track_id;
     }
     if (IsObjectInTheLane((*objects)[i], egolane_image, egolane_ground,
-                          &distance)) {
+                          world2camera, &distance) == true) {
```
Comment (xiaoxq) [inline modules/perception/camera/app/cipv_camera.cc]: `== true` is unnecessary.

File: modules/perception/camera/app/cipv_camera.cc [hunk 2]
```diff
@@ -720,26 +725,30 @@ bool Cipv::IsObjectInTheLaneGround(const std::shared_ptr<base::Object> &object,
 // Check if the object is in the lane in ego-ground space
 bool Cipv::IsObjectInTheLane(const std::shared_ptr<base::Object> &object,
                              const EgoLane &egolane_image,
-                             const EgoLane &egolane_ground, float *distance) {
-  if (b_image_based_cipv_) {
+                             const EgoLane &egolane_ground,
+                             const Eigen::Affine3d world2camera,
+                             float *distance) {
+  if (b_image_based_cipv_ == true) {
```
Comment (xiaoxq) [inline modules/perception/camera/app/cipv_camera.cc]: ditto

File: modules/perception/camera/tools/offline/visualizer.h [hunk 3]
```diff
@@ -49,13 +54,15 @@ class Visualizer {
   bool SetDirectory(const std::string &path);
   void ShowResult(const cv::Mat &img, const CameraFrame &frame);
   void Draw2Dand3D(const cv::Mat &img, const CameraFrame &frame);
-  void ShowResult_all_info_single_camera(
-      const cv::Mat &img, const CameraFrame &frame,
-      const base::MotionBufferPtr motion_buffer);
+  void ShowResult_all_info_single_camera(const cv::Mat &img,
+                                     const CameraFrame &frame,
```
Comment (xiaoxq) [inline modules/perception/camera/tools/offline/visualizer.h]: Align with first parameter.

File: modules/perception/camera/tools/offline/visualizer.h [hunk 4]
```diff
@@ -49,13 +54,15 @@ class Visualizer {
...
+                                     const base::MotionBufferPtr motion_buffer,
+                                     const Eigen::Affine3d &world2camera);
   void Draw2Dand3D_all_info_single_camera(const cv::Mat &img,
                                           const CameraFrame &frame,
                                           Eigen::Matrix3d intrinsic,
-                                          Eigen::Matrix4d extrinsic);
+                                          Eigen::Matrix4d extrinsic,
```
Comment (xiaoxq) [inline modules/perception/camera/tools/offline/visualizer.h]: Could they be const references?

File: modules/perception/camera/app/cipv_camera.cc (hunk 1 above)
Comment (techoe) [reply modules/perception/camera/app/cipv_camera.cc]: Thank you for the review! I removed "== true" in other places also.

File: modules/perception/camera/app/cipv_camera.cc (hunk 2 above)
Comment (techoe) [reply modules/perception/camera/app/cipv_camera.cc]: Handled.

File: modules/perception/camera/tools/offline/visualizer.h (hunk 3 above)
Comment (techoe) [reply modules/perception/camera/tools/offline/visualizer.h]: Aligned.

File: modules/perception/camera/tools/offline/visualizer.h (hunk 4 above)
Comment (techoe) [reply modules/perception/camera/tools/offline/visualizer.h]: Yes. I changed them to const.

**Pull request merged**